finagle-thrift/src/main/ruby/lib/finagle-thrift/trace.rb

```diff
@@ -57,6 +57,10 @@ def tracer=(tracer)
     @tracer = tracer
   end
 
+  def stack=(stack)
```

Comment:
We handle it by requiring that you use twitter's util libraries if you switch thread in java, but it won't be so simple in ruby.  I think for now we should just document the behavior, and then we can deal with it later if you actually need it.  Explicitly passing the trace data will probably be fine as a first approximation.

Grabbing a global mutex would require that you only serve one request at a time, which seems like it defeats the purpose.